neqo-transport/src/packet/mod.rs

```diff
@@ -839,14 +860,18 @@ impl<'a> PublicPacket<'a> {
             key_phase,
             pn,
             hdrbytes,
-            &self.data[self.header_len + pn_len..],
+            self.header_len + pn_len..self.data.len(),
         ))
     }
 
     /// # Errors
     ///
     /// This will return an error if the packet cannot be decrypted.
-    pub fn decrypt(&self, crypto: &mut CryptoStates, release_at: Instant) -> Res<DecryptedPacket> {
+    pub fn decrypt(
```

Comment:
I'm looking at the calling code and I *think* that we can change this to take `self` instead of `&mut self`.  There is some use of the public packet after decryption, but it seems like the necessary pieces can be moved to the decrypted packet.

Reply:
I'd suggest a separate PR for this.